src/Heap.js

```diff
@@ -35,11 +61,11 @@ export default class Heap {
         const leftChild = 2 * position + 1;
         const rightChild = 2 * position + 2;
             
-        if (leftChild < this.heap.length && this.compare(this.heap[leftChild], this.heap[position]) < 0) {
+        if (leftChild < this.heap.length && this.compare(this.heap[leftChild].key, this.heap[position].key) < 0) {
```

Comment:
would prefer checking `this.heap[smallest].key` for consistency with the check below